### ISSUE TITLE
parallel-map reads bad data for large payloads sometimes

### ISSUE THREAD

**Issue by erg** [factor/factor#2771]
https://paste.factorcode.org/paste?id=4547

I noticed this when reading two large json docs from servers that were not Factor -- sometimes the docs would be corrupted. The pastebin shows a test case that works in the same Factor instance.

Both the `all-eq?` check and the lengths are wrong sometimes. On macOS, if you single-step, when you `read-contents` and get to the first check for `stream-length` in:

```factor
M: input-stream stream-contents*
    (stream-contents-by-length-or-block) ; inline

: (stream-contents-by-length-or-block) ( stream -- seq )
    dup stream-length
    [ (stream-contents-by-length) ]
    [ (stream-contents-by-block) ] if* ; inline
```

it should get `f` because it's a tcp socket. However, on macOS and only when single-stepping, it gets a value just shy of 2 million (`fstat64` on the fd) which causes the wrong code path and terminates the test with only that many chars. On Windows, `stream-length` correctly returns `f` yet the test case still crosses the payloads.

On Linux, the length returns `f` as it should, even when single-stepping. The bug is also present.

When running the test with `100` in parallel, it is correct only 40-60% of the time on some runs.

**Comment by erg** [comment]
this is correct in .95, .97, and .98 -- a regression in master branch since .98!

**Comment by mrjbq7** [comment]
Maybe fixed in ce31bc9ead1ee5c37e91339a9892ec316bbe9a85.
